src/coreclr/src/gc/gc.cpp

```diff
@@ -26147,7 +26147,7 @@ void gc_heap::gc_thread_stub (void* arg)
 
     // server GC threads run at a higher priority than normal.
     GCToOSInterface::BoostThreadPriority();
-    _alloca (256*heap->heap_number);
+    void* tmp = _alloca (256*heap->heap_number);
```

Comment:
I think this is because we have that warning disabled: https://github.com/dotnet/runtime/blob/87809d68e77ab762691ec2472416d26dcd2a77da/eng/native/configurecompiler.cmake#L321
I tried using `(void)` workaround, it also fails like:

```sh
  [ 73%] Building C object src/jit/CMakeFiles/clrjit_win_x86_x64.dir/__/__/version.c.o
  In file included from /runtime/src/coreclr/src/gc/gcsvr.cpp:29:
  /runtime/src/coreclr/src/gc/gc.cpp: In static member function ‘static void SVR::gc_heap::gc_thread_stub(void*)’:
  /runtime/src/coreclr/src/gc/gc.cpp:26150:19: error: ignoring return value of ‘void* __builtin_alloca(long unsigned int)’ declared with attribute ‘warn_unused_result’ [-Werror=unused-result]
  26150 |     (void)_alloca (256*heap->heap_number);
  Scanning dependencies of target crossgen
```

another option is that we move `unused-result` suppression from https://github.com/dotnet/runtime/blob/b1ac8acf834812460972505b6860d00d66988dd5/src/coreclr/src/pal/src/CMakeLists.txt#L93-L95 to configurecompiler.cmake.

Reply:
Ok, let's leave your change as is

Reply:
It seems we should come up with some way to explicitly ignore return value that we can use without having unused result warning disabled. It seems that using a wrapper template method that just returns the passed in value and casting the result to (void) would work, as it would basically scrape off the attribute applied to the called method that causes the inability to use the cast to (void) directly.

Reply:
Agreed. I read the related discussion in this thread https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66425. This macro seems to be working reliably clang, gcc and msvc:

```c++
#define UNUSED(value) (void)!value
```
https://godbolt.org/z/93v7Ye

If it makes sense, I can do a followup PR and cleanup `unused-variable` and `unused-result` violations and suppression (as that `UNUSED` macro takes care of both).

Reply:
That would be great. The only thing is that we should make it after .NET 5 RC2 is out to minimize the churn in case we need to make some last minute changes.